training data shape (96000, 2968, 3)
reduced training data shape (3840, 2968, 3)
******************** Starting KPCA ********************
Traceback (most recent call last):
  File "/data/nvanhoorn/A3F_newstructure/VCBC_A3F_GLU_mut/redoing_everything/analysis/kernel_PCA/final_scripts/sigmoid_kernel/sigmoid.py", line 62, in <module>
    sigmoid_kernel_pca = kernel_pca_sigmoid.fit(train_data)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cm/shared/apps/anaconda3/python3.12/lib/python3.12/site-packages/sklearn/base.py", line 1389, in wrapper
    return fit_method(estimator, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cm/shared/apps/anaconda3/python3.12/lib/python3.12/site-packages/sklearn/decomposition/_kernel_pca.py", line 444, in fit
    self._fit_transform_in_place(K)
  File "/cm/shared/apps/anaconda3/python3.12/lib/python3.12/site-packages/sklearn/decomposition/_kernel_pca.py", line 368, in _fit_transform_in_place
    self.eigenvalues_ = _check_psd_eigenvalues(
                        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/cm/shared/apps/anaconda3/python3.12/lib/python3.12/site-packages/sklearn/utils/validation.py", line 2096, in _check_psd_eigenvalues
    raise ValueError(
ValueError: There are significant negative eigenvalues (0.233366 of the maximum positive). Either the matrix is not PSD, or there was an issue while computing the eigendecomposition of the matrix.
